pennylane/qchem/molecule.py

```diff
@@ -97,12 +102,21 @@ def __init__(
 
         self.symbols = symbols
         self.coordinates = coordinates
+        self.unit = unit
         self.charge = charge
         self.mult = mult
         self.basis_name = basis_name.lower()
         self.name = name
         self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols, load_data)
 
+        if self.unit == "Angstrom":
+            self.coordinates = self.coordinates / bohr_angs
+        elif self.unit not in ("Angstrom", "Bohr"):
+            raise ValueError(
+                f"The provided unit, '{unit}' is not supported. "
+                f"Please set 'unit' to 'Bohr' or 'Angstrom'."
+            )
```

Comment:
```suggestion
        if self.unit not in ("angstrom", "bohr"):
            raise ValueError(
                f"The provided unit, '{unit}' is not supported. "
                f"Please set 'unit' to 'Bohr' or 'Angstrom'."
            )
            
        if self.unit == "angstrom":
            self.coordinates = self.coordinates / bohr_angs
```

We should raise errors early.